service/middleware/header.go

```diff
@@ -70,6 +64,7 @@ func (h *Header) MiddlewareFunc(handler rest.HandlerFunc) rest.HandlerFunc {
 				}
 
 				if len(newFields) > 0 {
+					defer service.SetRequestLogger(request, oldLogger)
```

Comment:
so this wont be `nil` in this case?

Reply:
`request` and `oldLogger` can't be `nil` due to previous code that only executes this code if they are both non-`nil`.